xmtp_mls/src/groups/mls_sync.rs

```diff
@@ -78,6 +78,83 @@ use xmtp_proto::xmtp::mls::message_contents::plaintext_envelope::v2::MessageType
     Reply, Request,
 };
 
+#[derive(Debug, Error)]
+pub enum GroupMessageProcessingError {
+    #[error("[{0}] already processed")]
+    AlreadyProcessed(u64),
+    #[error("diesel error: {0}")]
+    Diesel(#[from] diesel::result::Error),
+    #[error("[{message_time_ns:?}] invalid sender with credential: {credential:?}")]
+    InvalidSender {
+        message_time_ns: u64,
+        credential: Vec<u8>,
+    },
+    #[error("invalid payload")]
+    InvalidPayload,
+    #[error("storage error: {0}")]
+    Storage(#[from] crate::storage::StorageError),
+    #[error(transparent)]
+    Identity(#[from] IdentityError),
+    #[error("openmls process message error: {0}")]
+    OpenMlsProcessMessage(#[from] openmls::prelude::ProcessMessageError),
+    #[error("merge staged commit: {0}")]
+    MergeStagedCommit(#[from] openmls::group::MergeCommitError<sql_key_store::SqlKeyStoreError>),
+    #[error("TLS Codec error: {0}")]
+    TlsError(#[from] TlsCodecError),
+    #[error("unsupported message type: {0:?}")]
+    UnsupportedMessageType(Discriminant<MlsMessageBodyIn>),
+    #[error("commit validation")]
+    CommitValidation(#[from] CommitValidationError),
+    #[error("epoch increment not allowed")]
+    EpochIncrementNotAllowed,
+    #[error("clear pending commit error: {0}")]
+    ClearPendingCommit(#[from] sql_key_store::SqlKeyStoreError),
+    #[error("Serialization/Deserialization Error {0}")]
+    Serde(#[from] serde_json::Error),
+    #[error("intent is missing staged_commit field")]
+    IntentMissingStagedCommit,
+    #[error("encode proto: {0}")]
+    EncodeProto(#[from] prost::EncodeError),
+    #[error("proto decode error: {0}")]
+    DecodeProto(#[from] prost::DecodeError),
+    #[error(transparent)]
```

Comment:
question: in previous code we have this comment:
`    #[error("intent error: {0}")]
    Intent(#[from] IntentError),`
    is this change intentional?